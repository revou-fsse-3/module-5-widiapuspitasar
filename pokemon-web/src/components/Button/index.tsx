import React, { useState } from "react";

interface ButtonProps {
    label: string;
    onClick: () => void;
    className?: string;
}

const Button = ({ label, onClick, className }: ButtonProps) => {
    const [buttonStyle, setButtonStyle] = useState({
        backgroundColor: 'black',
        color: 'white',
    });

    const hoverStyle = {
        backgroundColor: 'orange',
    };

    return (
        <button
            className={`bg-black color-white text-white ${className}`}
            style={buttonStyle}
            onClick={onClick}
            onMouseOver={() => setButtonStyle({ ...buttonStyle, ...hoverStyle })}
            onMouseOut={() => setButtonStyle({ backgroundColor: 'black', color: 'white' })}
        >
            {label}
        </button>
    );
};

export default Button;
