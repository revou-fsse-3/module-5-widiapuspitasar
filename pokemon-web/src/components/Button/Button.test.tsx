import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '.';
import "@testing-library/jest-dom/extend-expect"

describe('Button Component', () => {

  test('calls onClick callback when button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByText } = render(<Button label="Next" onClick={onClickMock} />);
    const buttonElement = getByText('Next');

    fireEvent.click(buttonElement);

    expect(onClickMock).toHaveBeenCalledTimes(1);
    expect(buttonElement).toMatchSnapshot();
  });

  test('applies additional className to the button', () => {
    const { getByText } = render(<Button label="Next" className="custom-button" onClick={function (): void {
        throw new Error('Function not implemented.');
    } } />);
    const buttonElement = getByText('Next');

    expect(buttonElement).toHaveClass('custom-button');
    expect(buttonElement).toMatchSnapshot();
  });
});
