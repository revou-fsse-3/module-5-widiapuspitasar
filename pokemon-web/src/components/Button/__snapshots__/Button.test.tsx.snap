// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component applies additional className to the button 1`] = `
<button
  class="bg-black color-white text-white custom-button"
  style="background-color: black; color: white;"
>
  Next
</button>
`;

exports[`Button Component calls onClick callback when button is clicked 1`] = `
<button
  class="bg-black color-white text-white undefined"
  style="background-color: black; color: white;"
>
  Next
</button>
`;
